src/plugins/data/server/search/routes/msearch.ts

```diff
@@ -67,16 +67,23 @@ export function registerMsearchRoute(router: IRouter, deps: SearchRouteDependenc
             })
           ),
         }),
+        query: schema.object({
+          data_source_id: schema.maybe(schema.string()),
+        }),
```

Comment:
Not very sure if the validation will give error if the request does not have any query.

Reply:
I've tested with data source disabled case. The `msearch` API works as expected without the data source id query.